src/bucket/test/BucketTests.cpp

```diff
@@ -399,6 +399,37 @@ TEST_CASE("merges proceed old-style despite newer shadows",
     }
 }
 
+TEST_CASE("lowest level merge converts live to init", "[bucket][bucketinit]")
```

Comment:
I think this test would be better as a BucketList level test instead of a `Bucket` level test. While I'm confident your code works today, there's an assumption here that `keepDeadEntries = false` is always the case and only the case for the bottom level bucket. While this is true today, it's an unstated assumption that could change in the future (as in will change with my state archival stuff), so a higher level test would test that:

1. Your code works (helpful for this PR)
2. This assumption in the BucketList logic is upheld (useful for future PRs)